### PR TITLE
Adds C++ CONBus Library for use with Arduino

### PR DESCRIPTION
**Pull request by noahzemlin**
- Supports reading and writing bytes to buffers that are implemented using static variables
- Example included of using CONBus using IGVC CAN Spec version 20230406
- Unit tested
- Arduino Library compatible with keyword highlighting

### CODE REVIEW

File: examples/CAN_Example/CAN_Example.ino [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    register_to_fetch = readRegisterMessage.registerAddress;
+  }
+
+  // CONBus write register
+  if (frame.id == (1200 + DEVICE_ID)) {
+
+    writeRegisterMessage = *(*CAN_writeRegisterMessage)frame.data;
+    conbus.writeRegisterBytes(writeRegisterMessage.registerAddress, writeRegisterMessage.value, writeRegisterMessage.length);
+  
+    memcpy(&writeRegisterResponseMessage, *writeRegisterMessage, sizeof(writeRegisterResponseMessage));
```
Comment (noahzemlin) [inline examples/CAN_Example/CAN_Example.ino]: Need to set awaiting_write_response to true

Comment (noahzemlin) [reply examples/CAN_Example/CAN_Example.ino]: Also, probably shouldn't be doing memcpy in interrupts like this even if it is only <= 8 bytes

File: library.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name=CON-Bus Lib
+version=0.1.0
+author=Sooner Robotics <scr@ou.edu>
+maintainer=Noah Zemlin <noah.zemlin@ou.edu>
+sentence=Implements the Sooner Robotics CON-Bus protocol
+paragraph=Documentation available on our GitHub wiki at https://github.com/SoonerRobotics/RobotLib/wiki. Made with love by those nerds at Sooner Competitive Robotics.
```
Comment (noahzemlin) [inline library.properties]: not robotlib

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Contributing to Testing
+
+## Catch2
+We use Catch2 for our unit testing framework. You can read more about it here https://github.com/catchorg/Catch2, but you probably can learn it through the examples in this directory alone.
+
+## Adding new tests
+If you would like to contribute new tests to new classes, simply create a new file called `test_NAME.cpp` where NAME is relevant to whatever class or function you are testing. Look at the other examples in this directory as a formatting guide. You should only need the following two imports
+```cpp
+#include <RobotLib.h>
```
Comment (noahzemlin) [inline tests/README.md]: not robotlib

**Comment by dylanzemlin** [comment]
Looks good to me. Adding a CAN driver would definitely be an interesting idea, and really nice for projects that use multiple firmware devices to prevent rewriting the same code. 😄

**Review by dylanzemlin — APPROVED** [review]

**Comment by noahzemlin** [comment]
Added a simple CAN Bus driver and rewrote the example to use it

**Pull request merged**